### DISCUSSION TITLE
Add support for selecting and traversing full function calls

### DISCUSSION THREAD

**Discussion by cezhunter** [helix-editor/helix#12256] [Ideas]
Hi everyone - this idea is two fold:
A) add a new match for function calls (including the parenths and args). For example
```
func(a, b)
something.func(c)
```
wherever the cursor is somewhere on the func call, I should be able to easily select the whole thing like so (bracket is selection):
```
[func(a,b)]
something.[func(c)].another()
```
Right now, it is not possible in one swift motion - requires multiple keystrokes and select mode.

B) I should be able to use the bracket menus to goto next/prev function calls - this is especially useful when dealing with function call chains:
```
something.func_a().func_b(1,2).func_c("wow")
```
Ideally I can do `[-goto-prev-call` when my cursor is at the end of the line and am able to select entire calls down the chain (function name, parenths, and arguments) and can perform `d`, `c`, etc on them.

The syntax tree traversal does not really allow for either of these cases.

**Comment by cezhunter** [comment]
Thank you to @NikitaRevenco for his PR #12254 which kinda touches on this concept.

**Comment by adaszko** [comment]
I'm also looking for a way to select a function name under the cursor, the opening parenthesis, and the closing parenthesis to operate on it as one object (multiple cursors).  I hardly ever need to operate on an entire function ([`f`](https://docs.helix-editor.com/textobjects.html)) but all the time on function *calls*.  Particularly, wrapping and unwrapping a selection in a function call.
